### PR TITLE
Migrate to angular v18

### PR DESCRIPTION
**Pull request by einfach3**
- Enabled the option to use the new build system during the migration, see here for more information:
https://angular.dev/tools/cli/build-system-migration

### CODE REVIEW

File: ui/angular.json [hunk 1]
```diff
@@ -9,13 +9,16 @@
       "projectType": "application",
       "architect": {
         "build": {
-          "builder": "@angular-devkit/build-angular:browser",
+          "builder": "@angular-devkit/build-angular:application",
```
Comment (einfach3) [inline ui/angular.json:12]: Switching to the new build system

File: ui/tsconfig.json [hunk 2]
```diff
@@ -4,11 +4,11 @@
     "baseUrl": "./",
     "outDir": "./dist/out-tsc",
     "forceConsistentCasingInFileNames": true,
+    "esModuleInterop": true,
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
     "sourceMap": true,
     "declaration": false,
-    "downlevelIteration": false,
```
Comment (einfach3) [inline ui/tsconfig.json:11]: Automatically removed during migration.
Also see: https://www.typescriptlang.org/tsconfig/#downlevelIteration

File: ui/tsconfig.json [hunk 3]
```diff
@@ -4,11 +4,11 @@
     "baseUrl": "./",
     "outDir": "./dist/out-tsc",
     "forceConsistentCasingInFileNames": true,
+    "esModuleInterop": true,
```
Comment (einfach3) [inline ui/tsconfig.json:7]: Automatically added during migration.
Also see: https://www.typescriptlang.org/tsconfig/#esModuleInterop

File: ui/src/app/app.module.ts [hunk 4]
```diff
@@ -9,20 +9,20 @@ import { AppRoutingModule } from './app-routing.module';
...
+  bootstrap: [AppComponent],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
   imports: [
     BrowserModule,
     MatButtonModule,
     AppRoutingModule,
     FeaturesModule,
     MatToolbarModule,
     ComponentsModule,
-    HttpClientModule,
```
Comment (einfach3) [inline ui/src/app/app.module.ts:23]: Is now deprecated

**Review by SebaRenner — APPROVED** [review]
LGTM!

**Pull request merged**